internal/worker/code_execution.go

```diff
@@ -0,0 +1,86 @@
+package worker
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/ossf/package-analysis/internal/featureflags"
+	"github.com/ossf/package-analysis/internal/sandbox"
+	"github.com/ossf/package-analysis/pkg/api/pkgecosystem"
+)
+
+// sandboxExecutionLogPath is the absolute path of the execution log file
+// inside the sandbox. The file is used for code execution feature.
+const sandboxExecutionLogPath = "/execution.log"
+
+var nonSpaceControlChars = regexp.MustCompile("[\x00-\x08\x0b-\x1f\x7f]")
+
+func isExecutePhaseEnabled(ecosystem pkgecosystem.Ecosystem) bool {
+	if !featureflags.CodeExecution.Enabled() {
+		return false
+	}
+
+	switch ecosystem {
+	case pkgecosystem.PyPI:
+		return true
+	default:
+		return false
+	}
+}
+
+// initialiseExecutePhase copies an empty file to the sandbox, which becomes the execution log.
+func initialiseExecutePhase(ctx context.Context, sb sandbox.Sandbox) error {
+	tempFile, err := os.CreateTemp("", "")
+	if err != nil {
+		return fmt.Errorf("could not create empty execution log file in host: %w", err)
+	}
+
+	// file wasn't written to, so don't worry too much about close errors
+	_ = tempFile.Close()
+	tempPath := tempFile.Name()
+
+	if err := sb.CopyIntoSandbox(ctx, tempPath, sandboxExecutionLogPath); err != nil {
+		return fmt.Errorf("could not copy empty execution log file to sandbox: %w", err)
+	}
+
+	// file wasn't written to, so don't worry too much about remove errors
+	_ = os.Remove(tempPath)
+
+	return nil
+}
```

Comment:
Is this function, and the use of `execution.log` necessary any more for gating the logic?

Given that the phase is now explicit I suspect this capability is no longer needed.

Reply:
Yes it can be removed.